MailContainerTest/Providers/MailContainerStoreProvider.cs

```diff
@@ -0,0 +1,32 @@
+﻿using MailContainerTest.Data;
+using Microsoft.Extensions.Configuration;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace MailContainerTest.Providers
```

Comment:
Would have liked to do something with the built in DI of .NET to automatically pull the correct interface concretion internally to the DLL rather than creating this public class, but do to time constraints I went with this approach.

With more time I would have investigated a way to keep this logic internal so the user wouldn't need to worry about setting up the DI when consuming this DLL.